azext_iot/_params.py

```diff
@@ -306,8 +306,8 @@ def load_arguments(self, _):
             help="If this flag is set, then the command will overwrite the state of the destination hub."
         )
         context.argument(
-            "overwrite_file",
```

Comment:
assuming this param only applies to `az iot hub state`, I would move it to be under a with statement for "az iot hub state" rather than "iot hub"

Reply:
Furthermore, move the help for `az iot hub state` to azure_iot/iothub/params.py